Documents/Testing Policy/TestPolicy.tex

```diff
@@ -4,128 +4,308 @@
+	\caption{Android Studio Test Example}
+	\label{fig:android-test}
+  \end{figure}
+ Both UI and Unit tests are executed in the same manner for Android, the primary difference being that a UI test will need to run on a connected device or virtual machine.\\
+\indent In order to run a unit or UI test for Android, one must open the file containing the desired test(s) using Android Studio as in figure \ref{fig:android-test}. Once there, one can right click on the green `test' button in the margin, then they can select to run the tests, the results of which will be seen at the bottom of the IDE.
+
+\subsection{iOS}
+\subsubsection{XCTest}
+XCTest integrates seamlessly into Xcode's (the iOS IDE) testing workflow, and for that reason is used to facilitate all testing on iOS. XCTest is used for both UI and unit testing on iOS.\\
+\indent iOS unit tests can be found \href{https://github.com/BinaryNinjaz/COS301-Capstone/tree/master/Source/iOS/Harvest/HarvestTests}{here}, and UI tests can be found \href{https://github.com/BinaryNinjaz/COS301-Capstone/tree/master/Source/iOS/Harvest/HarvestUITests}{here}.
+\subsubsection{Execution}
+To begin tests in Xcode, click on the play button in the top right, on the line that is highlighted in blue; as seen in figure \ref{fig:is}. This will run the tests. The results will look like figure \ref{fig:ie}.
+  \begin{figure}
+  	\includegraphics[width=\textwidth]{images/iOS-test.png}
+	\caption{iOS Xcode Test Start}
+	\label{fig:is}
+  \end{figure}
+
+\subsection{Web}
+\subsubsection{QUnit}
+QUnit is used for unit testing javascript functions. It was chosen since it displays its results in a convenient way---by including them in the web page, and it's very simple to use.\\
+\indent QUnit tests can be found \href{https://github.com/BinaryNinjaz/COS301-Capstone/tree/master/Source/Web/app}{here}.
+\subsubsection{Execution}
+\begin{figure*}[ht!]
+            \includegraphics[width=.5\textwidth]{images/q1.png}
+            \includegraphics[width=.5\textwidth]{images/q2.png}
+            \caption{Before Enabling Tests}
+            \label{fig:q1}
+            \caption{After Enabling Tests}
+            \label{fig:q2}
+        \end{figure*}
+\paragraph{QUnit} To execute QUnit tests, go into the html file that uses the functions to be tested. At the bottom, find a segment that mentions \texttt{QUnit Testing}, such as in figure \ref{fig:q1}; then simply uncomment the code, as in figure \ref{fig:q2}, and refresh the page. The results are now displayed, such as in figure \ref{fig:qr}.
+
+\begin{figure}
+\includegraphics[width=\textwidth]{images/qr.png}
+\caption{Results of QUnit test}
+\label{fig:qr}
+\end{figure}
+
+\section{Test Cases}
+\subsection{Database}
+The database cannot be tested locally with unit tests, so UI tests are used to perform create, read, update, and delete (CRUD) operations on the database. The tests also confirm that after the operations are completed, the database returns the correct information.\\
```

Comment:
It's true, but you can use dependancy injection to do unit tests with it see [here](https://github.com/BinaryNinjaz/COS301-Capstone/blob/525c530629e004b2ad3c382646f4d7f100349adc/Source/iOS/Harvest/HarvestTests/HarvestTests.swift#L205) downwards.

Reply:
FWICT you're still not testing the database itself, rather how the results it returns are handled.